### PR TITLE
Download functionality for the tool

### PR DESCRIPTION
**Pull request by matiasah**
I tried to add the download functionality to the tool, I'm not sure if it works, I couldn't test it because auto-refresh wasn't working on my client. I was expecting you could make something better out of it, I copied part of the E2 code because I'm not really a gmod scripter. (Don't blame me for the red lines that you see there, SciTE did it).

### CODE REVIEW

**Comment by Xandaros** [comment]
Ref #208

**Comment by awilliamson** [comment]
Ref #12

**Comment by matiasah** [comment]
That seems far, though, you could at least use this until you decide to make a new editor. Just think of it, it was posted months ago.

**Comment by Xandaros** [comment]
Did you even look at your own code?

**Comment by awilliamson** [comment]
It seems that whilst writing the PR, you failed to look at any guidelines laid out regarding this project. Not only have you failed to notice the ones laid out, but also you deemed fit to write code which intrudes upon external addon's namespace. Lastly, you create PR ( with the goal of it being accepted ) without actually testing if anything you've 'copy-pasted' works?

**Comment by matiasah** [comment]
@Xandaros: Yes, I did. The code is quite unreadable, I couldn't find out how to put the received code into the editor so I left it like that.
@awilliamson: As I said, auto-refresh wasn't working for reasons that I don't know. I can't test if the code works without auto-refresh. And no, it's not exactly copy-pasted, I took a small part of the code and made it so it works like E2.

I was expecting to receive kind answers around here, also I think I'm not going to use this due to the fact that it didn't let me join a server on which I uploaded this content. It broke the server for no reason.

File: lua/weapons/gmod_tool/stools/starfall_processor.lua [hunk 1]
```diff
@@ -32,6 +32,43 @@ if SERVER then
...
+			WireLib.AddNotify(ply, "Invalid Starfall chip specified.", NOTIFY_ERROR, 7, NOTIFYSOUND_DRIP3)
+			return
+		end
+
+		if not IsValid(ply) or not ply:IsPlayer() then
+			error("Invalid player entity (wtf??). This should never happen. " .. tostring(ply), 0)
+		end
+
+		if not hook.Run("CanTool", ply, WireLib.dummytrace(targetEnt), "starfall_processor") then
+			WireLib.AddNotify(ply, "You're not allowed to download from this Expression (ent index: "..targetEnt:EntIndex()..").", NOTIFY_ERROR, 7, NOTIFYSOUND_DRIP3)
```
Comment (Xandaros) [inline lua/weapons/gmod_tool/stools/starfall_processor.lua:48]: I repeat: Have you even looked at your own code?

**Comment by davidsonbr** [comment]
"I'm not going to use this due to the fact that it didn't let me join a server on which I uploaded this content. It broke the server for no reason."

If it 'broke' your server what makes you think that it would not 'break' another? Did you do anything other than copy-pasting the code and replacing some of the references to Expression 2 with Starfall?

**Comment by matiasah** [comment]
@EpicG: No I didn't upload the modified code to that server.

@Xandaros: I repeat: Yes, I did look at my own code.

**Comment by davidsonbr** [comment]
Alright, so there are a few reasons why you are getting bad reception with this code:
- Incorrect codestyle
- Nil references / errors
  - Lines such as `WireLib.AddNotify(ply, "Invalid Starfall chip specified.", NOTIFY_ERROR, 7, NOTIFYSOUND_DRIP3)` are referencing objects that in this case are nil. Unless you include the file that an external variable is defined in, it will be nil and produce an error. So those references to "WireLib" will not work.
  - `WireLib.StarfallDownload(ply, ent, nil, true)` Unless you edited a file other than the one you supplied, WireLib does not have a StarfallDownload function, so you will get an attempt to call nil.
- Copy-pasting and hoping that it will work.
- Remnants of your copy-paste still there.
  - Line 43, `WireLib.AddNotify(ply, "You're not allowed to download from this Expression (ent index: "..targetEnt:EntIndex()..").", NOTIFY_ERROR, 7, NOTIFYSOUND_DRIP3)`
- Stating that you have not tested it.
  - Moreover stating that you have not tested it because it did not work.

**Pull request closed without merging**